### PR TITLE
Internal improvement: Org scoping

### PR DESCRIPTION
**Pull request by eggplantzzz**
Continue org-scoping the packages
  1. truffle-hdwallet-provider --> @truffle/hdwallet-provider
  2. truffle-provider --> @truffle/provider
  3. truffle-provisioner --> @truffle/provisioner
  4. truffle-external-compile --> @truffle/external-compile
  5. truffle-compile --> @truffle/compile-solidity
  6. truffle-compile-vyper --> @truffle/compile-vyper

### CODE REVIEW

**Comment by eggplantzzz** [comment]
Do we want to consider renaming @truffle/compile to @truffle/compile-solidity now that we are renaming packages?

**Review by gnidan — APPROVED** [review]
I'm into renaming to `@truffle/compile-solidity`!

File: packages/external-compile/package.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
+  "name": "@truffle/external-compile",
   "description": "Wrapper to compile Truffle contracts with arbitrary shell command",
   "license": "MIT",
   "author": "g. nicholas d'andrea <gnidan@users.noreply.github.com>",
   "homepage": "https://github.com/trufflesuite/truffle#readme",
-  "repository": "https://github.com/trufflesuite/truffle/tree/master/packages/truffle-external-compile",
+  "repository": "https://github.com/trufflesuite/truffle/tree/master/packages/external-compile",
   "bugs": {
     "url": "https://github.com/trufflesuite/truffle/issues"
   },
```
Comment (CruzMolina) [inline packages/external-compile/package.json]: needs one of these

```
"publishConfig": {
    "access": "public"
  },
```

File: packages/hdwallet-provider/package.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
   "description": "HD Wallet-enabled Web3 provider",
   "license": "MIT",
   "author": "Tim Coulter <tim.coulter@consensys.net>",
-  "homepage": "https://github.com/trufflesuite/truffle/tree/master/packages/truffle-hdwallet-provider#readme",
-  "repository": "https://github.com/trufflesuite/truffle/tree/master/packages/truffle-hdwallet-provider",
+  "homepage": "https://github.com/trufflesuite/truffle/tree/master/packages/hdwallet-provider#readme",
+  "repository": "https://github.com/trufflesuite/truffle/tree/master/packages/hdwallet-provider",
   "bugs": {
     "url": "https://github.com/trufflesuite/truffle/issues"
   },
```
Comment (CruzMolina) [inline packages/hdwallet-provider/package.json]: also needs

```
"publishConfig": {
    "access": "public"
  },
```

**Review by CruzMolina — CHANGES_REQUESTED** [review]
Almost there!

**Review by CruzMolina — APPROVED** [review]

**Review by fFurkanllci — COMMENTED** [review]
Yfl

**Pull request merged**